### PR TITLE
Feature-Add-reducers-and-actions

### PR DESCRIPTION
**Pull request by lincoln1883**
For this milestone i was able to complete the following task:
- Add react-redux  and redux/toolkit to the the project file.
- Add redux folder with the books and categories folders.
- Add books slice and categories slice.
- Import createSlice from redux/toolkit, and create intial state and actions.

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  initialState,
+  reducers: {
+    getCategories: (state) => {
+      state.categories = 'Under Construction';
+    },
+  },
+});
+
+export default categoriesSlice.reducer;
+export const { getCategories } = categoriesSlice.actions;
```
Comment (bobb-Rob) [inline src/redux/categories/categoriesSlice.js:18]: - [ ] In your `/src/redux/store.js` file, based on the project requirement, you are expected to create the store object using the `configureStore` method from the Redux/toolkit.  Kindly implement the following: 

    * Imports `configureStore `from `@reduxjs/toolkit`.
    * Imports `books` from `src/redux/books`.
    * Imports `categories` from `src/redux/categories`.
    * Use `configureStore` to create a reducer.
      * `books` and `categories` should be added to `configureStore` method.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @lincoln1883 , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (lincoln1883) [reply src/redux/categories/categoriesSlice.js:18]: thanks @bobb-Rob but that was not listed in the requirements sorry for the oversight.

Project requirements
Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
Create a directory that will contain all your Redux logic (/src/redux)
Configure a Redux store (/src/redux/store.js)
Define a slice of state for books that:
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book
Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:
/src
    |---/components
    |---/redux
        |--- /books
                | booksSlice.js
        |--- /categories
                | categoriesSlice.js
        | store.js
    | index.js
NOTE: No changes in React components are required in this project.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @lincoln1883👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**